save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/index/IndexViewAuth.kt

```diff
@@ -0,0 +1,114 @@
+/**
+ * Authorization component (Oauth2 elements) for Index View
+ */
+
+package com.saveourtool.save.frontend.components.views.index
+
+import com.saveourtool.save.frontend.externals.fontawesome.faCopyright
+import com.saveourtool.save.frontend.externals.fontawesome.faGithub
+import com.saveourtool.save.frontend.externals.fontawesome.faSignInAlt
+import com.saveourtool.save.frontend.externals.fontawesome.fontAwesomeIcon
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.noopResponseHandler
+import com.saveourtool.save.info.OauthProviderInfo
+
+import js.core.jso
+import org.w3c.fetch.Headers
+import react.ChildrenBuilder
+import react.FC
+import react.VFC
+import react.dom.html.ReactHTML.a
+import react.dom.html.ReactHTML.div
+import react.useState
+import web.cssom.ClassName
+import web.cssom.FontSize
+import web.cssom.rem
+
+import kotlinx.browser.window
+
+val indexAuth: FC<IndexViewProps> = FC { props ->
+    val (oauthProviders, setOauthProviders) = useState(emptyList<OauthProviderInfo>())
+    val getOauthProviders = useDeferredRequest {
```

Comment:
`useDeferredRequest` + `useOnce` could be replaced with `useRequest`

Reply:
I will need to call getOauthProviders() and I am afraid that outside of useOnce it can affect us

Reply:
ok